src/device/rttDevice.js

```diff
@@ -271,33 +240,27 @@ class RTTDevice extends Device {
         }
     }
 
-    static getHardwareStates(serialNumber) {
+    static getHardwareStates(deviceId) {
         let iteration = 0;
         async function readUntil() {
             let hwStates;
             while (!hwStates && iteration < 250) {
                 iteration += 1;
                 // eslint-disable-next-line
                 hwStates = await new Promise((resolve, reject) => {
-                    nRFjprogjs.rttRead(
-                        serialNumber,
-                        0,
-                        255,
-                        (err, string, rawbytes, time) => {
-                            if (err) {
-                                return reject(new Error(err));
-                            }
+                    nRFDeviceLib
+                        .rttRead(deviceLibContext, deviceId, 0, 255)
+                        .then(({ rttReadBuffer: rawbytes }) => {
                             if (rawbytes.length) {
                                 return resolve({
-                                    string,
                                     rawbytes,
-                                    time,
                                     iteration,
+                                    string: rawbytes.toString(),
                                 });
                             }
                             return resolve();
-                        }
-                    );
+                        })
+                        .catch(err => reject(new Error(err)));
```

Comment:
Just like above, instead of wrapping the call to `nRFDeviceLib.rttRead` in a Promise you can use that it itself already returns a Promise now.